pkg/cmd/operator/webhooks_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2021 ScyllaDB
+
+package operator
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net/http"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+)
+
+const (
+	pollInterval = 1 * time.Second
+	pollTimeout  = 3 * time.Second
+)
+
+func TestMain(m *testing.M) {
+	dynamiccertificates.FileRefreshDuration = 1 * time.Second
+	os.Exit(m.Run())
+}
+
+func TestRunErrors(t *testing.T) {
+	tt := []struct {
+		Name           string
+		WebhookOptions *WebhookOptions
+		ExpectedError  error
+	}{
+		{
+			Name: "invalid port",
+			WebhookOptions: &WebhookOptions{
+				Port:                           65536,
+				InsecureGenerateLocalhostCerts: true,
+			},
+			ExpectedError: fmt.Errorf("can't create listener: listen tcp: address 65536: invalid port"),
+		},
+	}
+
+	for i := range tt {
+		test := tt[i]
+		t.Run(test.Name, func(t *testing.T) {
+			t.Parallel()
+
+			var wg sync.WaitGroup
+			defer wg.Wait()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			err := test.WebhookOptions.Complete()
+			if err != nil {
+				t.Fatalf("can't complete WebhookOptions: %v", err)
+			}
+
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				err := test.WebhookOptions.run(ctx, genericclioptions.IOStreams{})
+				if !reflect.DeepEqual(test.ExpectedError, err) {
+					t.Errorf("expected and actual errors differ: %s", cmp.Diff(test.ExpectedError, err))
+				}
+			}()
+		})
+	}
+}
+
+func TestRunWebhookServerCertReload(t *testing.T) {
+	var wg sync.WaitGroup
+	defer wg.Wait()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dir := t.TempDir()
+	tlsCertFile := dir + "/tls.crt"
+	tlsKeyFile := dir + "/tls.key"
+
+	cc, err := newCertificateCreator()
+	if err != nil {
+		t.Fatalf("failed to create CertificateCreator: %v", err)
+	}
+
+	initialCertSerialNumber := big.NewInt(1)
+	encodedCert, err := cc.generateEncodedCertificate(initialCertSerialNumber)
+	if err != nil {
+		t.Fatalf("failed to create new encoded certificate: %v", err)
+	}
+	if err = createFileWithContent(tlsCertFile, encodedCert.cert); err != nil {
```

Comment:
nit: this is pretty long, it might be clearer to initialize the variable on a dedicated line (hint: it always is)